app/src/main/res/values/styles.xml

```diff
@@ -39,6 +39,12 @@
 
     <style name="AppTheme.PopupOverlay" parent="AppTheme" />
 
+    <style name="AppTheme.Dialog" parent="Theme.Material3.DayNight.Dialog">
```

Comment:
> What's missing now is getting the permission. As it's rather optional to use this feature I wonder how we want to inform the user about the missing permission.
> 
> Same as with the other permissions right at the launch (InitializationActivity) or via the settings?

I'd say we don't add anything in addition, sure it may seem a little tedious to always confirm the dialog, but I don't think this feature is frequently used.

Reply:
I don't think we are about about the same thing here xD  
What I mean is this permission right here:  
https://github.com/cyb3rko/android/blob/7016d28804988cac53b7fd7f8359d3db614a8e54/app/src/main/AndroidManifest.xml#L13

It seems like the Play Store activates it by default, but a side-install will lead to this permission being declined by default. So if people want to use this feature, they would have to find this permission and manually allow it in the settings. Even took me some minutes to figure it out.

Otherwise the dialog is not allowed to show up on the screen at any time.

Reply:
According to this comment (https://stackoverflow.com/a/36019034/9077356) from 2016, I don't know if it still applies, this applies to every install aside from the Play Store.

So we include that into the settings to allow permission?

Reply:
I've installed the gotify apk manually by downloading it to my phone and also via ADB. The dialog works fine without any additional permission checks (tested on Android 13).

Is this SYSTEM_ALERT_WINDOW permission even required for the dialog?

Reply:
Does it work even if you close the app before sending the notification?

Reply:
No, but this is a documented limitation of this feature:

> onReceive.intentUrl
> Only works when the gotify app is in focus (limitation of android)

https://gotify.net/docs/msgextras#androidaction

So I guess it's good enough as is?

Reply:
It's up to you. I think we should at least tell users that it's possible outside of the app when giving the permission (which is seemingly already defined in the manifest, so no new permission here).
I feel like that would improve usability of that feature as your not required to have the app opened.

Reply:
Done. Something like this?

Reply:
Yeah, looks good.

Reply:
Then maybe we should add that to the documentation.  
Do we need an indicator for the setting to instantly see if the permission has been given?

Reply:
Adding it to the docs sounds good, I can do that. Well, I don't think that we need it, but if you want to implement it go ahead (:.

Could you rebase your branch so that the fix blocking permissions commit will be removed?
Maybe you could also squash maybe one for implementing the dialog and one for adding the preference?

Reply:
> So just two commits on this branch and drop the merged commit and code?

Yes, doing a `git rebase master` (when master is up-to-date) should automatically remove the merge commit with its changes.

Reply:
That should be it

Reply:
Have you added it to the docs? I could not find it yet

Reply:
Ahh, thanks for the reminder (:. added it with gotify/website@f403abf2d0005067e280d93cb69855be1454ea96.

Reply:
Awesome 👍🏻